app/routes/task_routes.py

```diff
@@ -1 +1,91 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response
+from ..db import db
+from .route_utilities import validate_model, create_model_from_dict
+from app.models.task import Task
+from datetime import datetime
+import os, requests
+
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix = "/tasks")
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+    return create_model_from_dict(Task, request_body)
+
+
+@tasks_bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    sort_param = request.args.get("sort")
+
+    if sort_param == "asc":
+        query = query.order_by(Task.title.asc())
+    
+    if sort_param == "desc":
+        query = query.order_by(Task.title.desc())
+
+    else:
+        query = query.order_by(Task.id)
+        
+    tasks = db.session.scalars(query).all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+    return tasks_response
+
+@tasks_bp.get("<task_id>")
+def get_task_by_id(task_id):
+    task = validate_model(Task, task_id)
+
+    return {"task": task.to_dict()}
+
+@tasks_bp.put("<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return Response(status=204, mimetype="application/json")
```

Comment:
This is another opportunity to D.R.Y. up our code! Notice how in this route and the `PUT` route for `goal_routes.py` we follow the pattern of:
```py
object.ATTRIBUTE = request_body["ATTRIBUTE"]
```
We use `hasattr` and [setattr](https://docs.python.org/3/library/functions.html#setattr) to make a helper function to update our `Task` and `Goal` model. It would look like this: 
```py
def update_model(obj, data):
    for attr, value in data.items():
        if hasattr(obj, attr):
            setattr(obj, attr, value)
    
    db.session.commit()

    return Response(status=204, mimetype="application/json")
```

This refactor not only makes our code D.R.Y but shows that we recognize logic that has higher level usability.